Add support for kube groups + Daemon Version Fix

sidpremkumar opened the pull request:
## Description of the change

In this PR: 
* Adds support for impersonating kube groups 
* Adds `npm run release-prod` function that will inject daemon version into our kube daemon

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Refactor (code change with no functionality change)
- [ ] Enhancement (minor change below the level of a feature)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Remove feature (removes a feature we don't support anymore)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related JIRA tickets

Relates to JIRA: CWC-1126, CWC-1094

## Relevant release note information

Release Notes:

## Potential Security Impacts

Does this PR have any security impact?
Does it fix a security issue?
Does it add attack surface?
Why do we think this change is safe?

## Checklists

### Development

- [ ] Does the code changed/added as part of this pull request have test coverage?
- [ ] Do all tests related to the changed code pass in development?
- [ ] Have you ensured that at least one other person has tested your code?
- [ ] Have you tested the code?
- [ ] Have you attached any pictures (if relevant)?

sidpremkumar commented:
![2021-09-28 at 4 46 AM](https://user-images.githubusercontent.com/28202162/135054951-d3a841dd-c91c-4651-9ad2-81ee804621e8.jpg)

tflpd commented on src/handlers/target-group/list-target-group.handler.ts:
```diff
@@ -0,0 +1,35 @@
+            await cleanExit(0, logger);
+        }
+        // regular table output
+        const tableString = getTableOfTargetUsers(targetGroups);
```
Let's create a new util for this or parameterize it with a bool flag. Otherwise these groups will be printed under the `Allowed Target Users` header - not ideal

sidpremkumar replied:
https://github.com/bastionzero/zli/pull/210/commits/1bedc09914a9d7126cb0b1257fb42caf5b6aa8f7

tflpd reviewed (approved):
LGTM; just two things:
- I saw `groups`, `clusterGroups`, `targetGroups`, `assumeGroups`. This will end up causing a lot of confusion, can we stick to one max two names? I would avoid using `groups` as they will be probably used later as `idpGroups` following the UI. Probably I would go with `targetGroups` following the pattern we have for `targetUsers` but this is your call, I understand why `clusterGroups` or `assumeGroups` sounds more reasonable  
- Mike asked me to bring back the kube commands on the help screen, let's remove the `description = false` from the commands you just added and the ones I did earlier please

sidpremkumar commented:
> I saw groups, clusterGroups, targetGroups, assumeGroups. This will end up causing a lot of confusion, can we stick to one max two names? I would avoid using groups as they will be probably used later as idpGroups following the UI. Probably I would go with targetGroups following the pattern we have for targetUsers but this is your call, I understand why clusterGroups or assumeGroups sounds more reasonable

Yes I agree, I will go through and change as much as I can. I want to move everything to `targetGroup` and `targetUser` eventually.

sidpremkumar commented on kube-go-assets/bctl/agent/datachannel/datachannel.go:
```diff
@@ -31,11 +31,13 @@ type DataChannel struct {
 
 func NewDataChannel(logger *lggr.Logger,
 	role string,
+	groups []string,
```
rename these to targetGroup/targetUser

sidpremkumar commented on kube-go-assets/bctl/agent/controlchannel/controlchannelmessages.go:
```diff
@@ -1,9 +1,10 @@
-	Role         string `json:"role"`
-	Token        string `json:"token"`
+	ConnectionId string   `json:"connectionId"`
+	Role         string   `json:"role"`
```
maybe this can go to targetRole / targetUser

sidpremkumar commented:
Updated tunnel screen using the standard table: 
![2021-10-01 at 5 00 PM](https://user-images.githubusercontent.com/28202162/135685620-3fdd05ac-2396-4a24-8e93-968cfc0a91f1.jpg)

sidpremkumar commented:
![2021-10-01 at 5 00 PM](https://user-images.githubusercontent.com/28202162/135685662-36e23992-f4dc-4549-87d6-f9f71a0d08ca.jpg)

ymarcus93 commented on src/handlers/target-group/add-target-group.handler.ts:
```diff
@@ -0,0 +1,63 @@
+        console.log(policy);
+        break;
+    default:
+        logger.error(`Adding target group to policy ${policyName} failed. Adding target users to ${policy.type} policies is not currently supported.`);
```
```suggestion
        logger.error(`Adding target group to policy ${policyName} failed. Adding target groups to ${policy.type} policies is not currently supported.`);
```

sidpremkumar replied:
https://github.com/bastionzero/zli/pull/210/commits/bda43e6e111d79eb9815e5976b11b0e17e508fa6

sidpremkumar commented:
@tflpd I noticed that `describe-cluster` was not working as intented: 
![2021-10-04 at 2 49 PM](https://user-images.githubusercontent.com/28202162/135907345-ccbb26b2-22c7-4f41-b367-83df4a1e7979.jpg)
First, you can see that the `targetUsers` are incorrect, those are the targetUsers available on the cluster not attached to the policy. 

So I went ahead and made the changes to it reflects the groups and targetUsers associated with each policy: 
![2021-10-04 at 2 50 PM](https://user-images.githubusercontent.com/28202162/135907492-9083a2ab-208b-4be2-b31a-34c17a0ae8da.jpg)

I've updated the name `describe-cluster` to `describe-cluster-policy`. I'm also planning on bring it up with Mike as well once I get the ok from you. 

Right now `describe-cluster` (other than the policy name) is not giving any more info than `zli lt -d`

tflpd commented:
> @tflpd I noticed that `describe-cluster` was not working as intented: ![2021-10-04 at 2 49 PM](https://user-images.githubusercontent.com/28202162/135907345-ccbb26b2-22c7-4f41-b367-83df4a1e7979.jpg) First, you can see that the `targetUsers` are incorrect, those are the targetUsers available on the cluster not attached to the policy.
> 
> So I went ahead and made the changes to it reflects the groups and targetUsers associated with each policy: ![2021-10-04 at 2 50 PM](https://user-images.githubusercontent.com/28202162/135907492-9083a2ab-208b-4be2-b31a-34c17a0ae8da.jpg)
> 
> I've updated the name `describe-cluster` to `describe-cluster-policy`. I'm also planning on bring it up with Mike as well once I get the ok from you.
> 
> Right now `describe-cluster` (other than the policy name) is not giving any more info than `zli lt -d`

- Oh, I don't think that was ever clarified as the req. Anyway, my bad, thanks for noticing! I see that you removed the environment column, did you mean to do that?  
- There are still plenty of places like the daemon or the zli that we use different names for the groups, if you just go to the file diff in the PR you will notice them. Approving this but if we leave them for another day this day will never come :(

sidpremkumar commented:
> > @tflpd I noticed that `describe-cluster` was not working as intented: ![2021-10-04 at 2 49 PM](https://user-images.githubusercontent.com/28202162/135907345-ccbb26b2-22c7-4f41-b367-83df4a1e7979.jpg) First, you can see that the `targetUsers` are incorrect, those are the targetUsers available on the cluster not attached to the policy.
> > So I went ahead and made the changes to it reflects the groups and targetUsers associated with each policy: ![2021-10-04 at 2 50 PM](https://user-images.githubusercontent.com/28202162/135907492-9083a2ab-208b-4be2-b31a-34c17a0ae8da.jpg)
> > I've updated the name `describe-cluster` to `describe-cluster-policy`. I'm also planning on bring it up with Mike as well once I get the ok from you.
> > Right now `describe-cluster` (other than the policy name) is not giving any more info than `zli lt -d`
> 
> * Oh, I don't think that was ever clarified as the req. Anyway, my bad, thanks for noticing! I see that you removed the environment column, did you mean to do that?

Yup so I did mean to get ride of env bc that can be seen in `zli lt -d`

> * There are still plenty of places like the daemon or the zli that we use different names for the groups, if you just go to the file diff in the PR you will notice them. Approving this but if we leave them for another day this day will never come :(

So yes I added this commit: https://github.com/bastionzero/zli/pull/210/commits/26bd8db2bb4d406ad9beb411f3fc3439d32e09cf. I compleatly agree, could you let me know if you see any other refs, trying to go with `targetUser` and `targetGroup`

tflpd commented:
> > > @tflpd I noticed that `describe-cluster` was not working as intented: ![2021-10-04 at 2 49 PM](https://user-images.githubusercontent.com/28202162/135907345-ccbb26b2-22c7-4f41-b367-83df4a1e7979.jpg) First, you can see that the `targetUsers` are incorrect, those are the targetUsers available on the cluster not attached to the policy.
> > > So I went ahead and made the changes to it reflects the groups and targetUsers associated with each policy: ![2021-10-04 at 2 50 PM](https://user-images.githubusercontent.com/28202162/135907492-9083a2ab-208b-4be2-b31a-34c17a0ae8da.jpg)
> > > I've updated the name `describe-cluster` to `describe-cluster-policy`. I'm also planning on bring it up with Mike as well once I get the ok from you.
> > > Right now `describe-cluster` (other than the policy name) is not giving any more info than `zli lt -d`
> > 
> > 
> > 
> > * Oh, I don't think that was ever clarified as the req. Anyway, my bad, thanks for noticing! I see that you removed the environment column, did you mean to do that?
> 
> Yup so I did mean to get ride of env bc that can be seen in `zli lt -d`
> 
> > * There are still plenty of places like the daemon or the zli that we use different names for the groups, if you just go to the file diff in the PR you will notice them. Approving this but if we leave them for another day this day will never come :(
> 
> So yes I added this commit: [26bd8db](https://github.com/bastionzero/zli/commit/26bd8db2bb4d406ad9beb411f3fc3439d32e09cf). I compleatly agree, could you let me know if you see any other refs, trying to go with `targetUser` and `targetGroup`

Just checked the diffs, I think we are all good with the renaming!
